src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala

```diff
@@ -16,44 +16,86 @@
 package nl.knaw.dans.easy.archivebag
 
 import java.io._
-import java.net.{URI, URL}
-import java.nio.file.{Files, Paths}
+import java.net.URI
+import java.nio.file.{Files, Path, Paths}
 import java.util.zip.{ZipEntry, ZipOutputStream}
 
 import org.apache.commons.codec.digest.DigestUtils
-import org.apache.commons.codec.net.URLCodec
-import org.apache.commons.io.IOUtils
+import org.apache.commons.io.{FileUtils, IOUtils}
 import org.apache.http.auth.{AuthScope, UsernamePasswordCredentials}
-import org.apache.http.client.methods.{CloseableHttpResponse, HttpGet, HttpPost, HttpPut}
+import org.apache.http.client.methods.{CloseableHttpResponse, HttpGet, HttpPut}
 import org.apache.http.entity.{ContentType, FileEntity}
 import org.apache.http.impl.client.{BasicCredentialsProvider, CloseableHttpClient, HttpClients}
 import org.slf4j.LoggerFactory
 
-import scala.util.{Failure, Try}
-import scala.xml._
+import scala.util.{Success, Try}
 
-object EasyArchiveBag {
+object EasyArchiveBag extends Bagit4FacadeComponent {
   val log = LoggerFactory.getLogger(getClass)
+  type BagId = String
+  val bagFacade = new Bagit4Facade()
 
   def main(args: Array[String]) {
     implicit val settings = CommandLineOptions.parse(args)
     run.get
   }
 
   def run(implicit ps: Parameters): Try[URI] = Try {
+    for {
```

Comment:
if something fails in this for-comprehension, you will never notice or catch it! Let's get together to fix this before merging!